### PR TITLE
Use async/await in AudioWeb class

### PR DESCRIPTION
**Pull request by nevik**
This replaces `Promise.then()` usage with async/await in the `AudioWeb` class, as part of #402.

@mikehenrty, please take a look at the remaining usages of `then()` in the web part of the code. I would actually leave them as they are, because they intentionally spawn off asynchronous actions and don't directly wait for them.
Introducing async/await in their place would pull up the Promise another call, and require changes in more places. In contrast to our goal with #402, this would actually degrade readability IMO.

### CODE REVIEW

**Comment by nevik** [comment]
@mikehenrty I removed the cached Promise from `AudioWeb` now (and made `init` in `AudioIOS` also async).

As we had discussed on IRC, this leaves certain edge cases where the user can confuse the system. An obvious one is quickly tapping/clicking the record button twice, since that can cause a race condition between two separate instances trying to initialize the microphone.

However, unlike what I had previously said to you, I think this cached Promise is not the right way to combat this problem. Even with it in place, there are other race conditions possible in `record.tsx` when it comes to starting a recording.

We should tackle this separately, and prevent multiple starting of a recording in `record.tsx` directly. This kind of mutexing/locking is IMO not part of the responsibility of `AudioWeb` and `AudioIOS`, since they're not intended to be shared among multiple consumers.

File: web/src/lib/components/pages/record/audio-web.ts [hunk 1]
```diff
@@ -111,68 +110,63 @@ export default class AudioWeb {
     this.volumeCallback = cb;
   }
 
-  init() {
-    if (this.initPromise) {
-      return this.initPromise;
+  async init(): Promise<void> {
```
Comment (mikehenrty) [inline web/src/lib/components/pages/record/audio-web.ts]: We still need a check here to see if we are already initialized. I agree we will still have an underlying race if the user clicks record twice really fast, and we should solve that in a different bug. But with this change, we initialize a new audio context every time the user presses record, which is unnecessary.

**Comment by nevik** [comment]
@mikehenrty alright, cached Promise is now replaced with a boolean flag as you suggested :)

File: web/src/lib/components/pages/record/audio-web.ts [hunk 2]
```diff
@@ -111,68 +111,67 @@ export default class AudioWeb {
...
+    if (this.isInitialized) {
```
Comment (mikehenrty) [inline web/src/lib/components/pages/record/audio-web.ts]: Where do we set `this.isInitialized` to true?

Comment (nevik) [reply web/src/lib/components/pages/record/audio-web.ts]: Ugh. Good thing you're paying attention!

I had overlooked that we already have that `ready` flag you mentioned previously, and then additionally failed to correctly implement the lifecycle for this new flag!

I've updated it to use the `ready` flag instead now, whose lifecycle should be fine.

**Comment by mikehenrty** [comment]
Great!

**Comment by nevik** [comment]
FINALLY 🎉

**Comment by mikehenrty** [comment]
it only took 1 month! 📆

**Comment by nevik** [comment]
The very image of fast and lean development!

**Pull request merged**